haystack_experimental/components/query/query_expander.py

```diff
@@ -0,0 +1,253 @@
+from haystack.dataclasses.chat_message import ChatMessage
+from haystack.utils.deserialization import deserialize_chatgenerator_inplace
+
+logger = logging.getLogger(__name__)
+
+
+DEFAULT_PROMPT_TEMPLATE = """
+You are part of an information system that processes user queries for retrieval.
+You have to expand a given query into {{ n_expansions }} queries that are
+semantically similar to improve retrieval recall.
+
+Structure:
+Follow the structure shown below in examples to generate expanded queries.
+
+Examples:
+1.  Query: "climate change effects"
+    {"queries": ["impact of climate change", "consequences of global warming", "effects of environmental changes"]}
+
+2.  Query: "machine learning algorithms"
+    {"queries": ["neural networks", "clustering techniques", "supervised learning methods", "deep learning models"]}
+
+3.  Query: "open source NLP frameworks"
+    {"queries": ["natural language processing tools", "free nlp libraries", "open-source NLP platforms"]}
+
+Guidelines:
+- Generate queries that use different words and phrasings
+- Include synonyms and related terms
+- Maintain the same core meaning and intent
+- Make queries that are likely to retrieve relevant information the original might miss
+- Focus on variations that would work well with keyword-based search
+- Respond in the same language as the input query
+
+Your Task:
+Query: "{{ query }}"
+
+You *must* respond with a JSON object containing a "queries" array with the expanded queries.
+Example: {"queries": ["query1", "query2", "query3"]}"""
+
+
+@component
+class QueryExpander:
+    """
+    A component that returns a list of semantically similar queries to improve retrieval recall in RAG systems.
+
+    Usage example:
+
+    ```python
+    from haystack.components.generators.chat.openai import OpenAIChatGenerator
+    from haystack_experimental.components.query import QueryExpander
+
+    expander = QueryExpander(
+        chat_generator=OpenAIChatGenerator(model="gpt-4o-mini"),
+        n_expansions=3
+    )
+
+    result = expander.run(query="green energy sources")
+    print(result["queries"])
+    # Output: ['alternative query 1', 'alternative query 2', 'alternative query 3', 'green energy sources']
+    # Note: Up to 3 additional queries + 1 original query (if include_original_query=True)
+
+    # To control total number of queries:
+    expander = QueryExpander(n_expansions=2, include_original_query=True)  # Up to 3 total
+    # or
+    expander = QueryExpander(n_expansions=3, include_original_query=False)  # Exactly 3 total
+    ```
+    """
+
+    def __init__(
+        self,
+        *,
+        chat_generator: Optional[ChatGenerator] = None,
+        prompt_template: Optional[str] = None,
+        n_expansions: int = 4,
+        include_original_query: bool = True,
+    ):
+        """
+        Initialize the QueryExpander component.
+
+        :param chat_generator: The chat generator component to use for query expansion.
```

Comment:
I would expand the `chat_generator` docstring, explaining the default `chat_generator` used if this parameter is not provided.